src/middleware/queries.ts

```diff
@@ -1690,3 +1690,72 @@ export function nftCollectionHolders(
     variables: { size: size?.toNumber(), start: start?.toNumber(), assetId },
   };
 }
+
+type MultiSigProposalQueryParameters = {
+  multisigId: string;
+};
+
+/**
+ *  @hidden
+ */
+export function createmultiSigProposalsQueryFilters(): {
+  args: string;
+  filter: string;
+} {
+  const args = ['$size: Int, $start: Int'];
+  const filters = [];
+
+  args.push('$multisigId: [String!]');
+  filters.push('multisigId: { eq: $multisigId }');
+
+  return {
+    args: `(${args.join()})`,
+    filter: `filter: { ${filters.join()} },`,
+  };
+}
+
+/**
+ * @hidden
+ *
+ * Get MultiSig Proposals history for a given MultiSig address
+ */
+export function multiSigProposalsQuery(
+  multisigId: string,
+  size?: BigNumber,
+  start?: BigNumber
+): QueryOptions<PaginatedQueryArgs<MultiSigProposalQueryParameters>> {
+  const { args, filter } = createmultiSigProposalsQueryFilters();
+
+  const query = gql`
+  query MultiSigProposalsQuery
```

Comment:
There is no need for separate query. Already we have `multiSigProposalQuery` in line 1525. You can just extend that to suffice our use case